### ISSUE TITLE
[Feature request] Add option to filter CodeLens links in BUILD files

### ISSUE THREAD

**Issue by rossc719g** [bazel-contrib/vscode-bazel#300]
TL;DR: It would be really nice to have a way to filter the displayed CodeLens links.

In BUILD files, there are often extra hidden rules defined by rules.

For example, if you use:
```
load("@bazel_skylib//rules:build_test.bzl", "build_test")

build_test(
  name="foo"
  ...
)
```
You get the expected `Test //whatever:foo` link, but you also get `Build //whatever:foo_0__deps`.

And if you have functions in a `.bzl` file that create multiple sub-rules, the list of links can quickly get very long, and end up wider than the available space, and the UI does not let you scroll right if your file content is not that wide.
(Incidentally, it would be nice to wrap these long lines of links, but that is a separate FR)

Ideally, it would just honor the existing `Bazel › Command Line: Query Expression` option for these links as well.
But it would be fine to have a similar option specifically for the CodeLens links if that is deemed to be nicer.

Thoughts?

**Comment by KrauseStefan** [comment]
This would be really nice to have both for the CodeLens links but also for the side bar tree.

The Idea plugin has this trough the `no-ide` tag.
https://ij.bazel.build/docs/tags.html#no-ide

Something similar could be very nice.
Alternatively just a setting to add filter by name.

What something like this be a welcome change for the maintainers?

**Comment by vogelsgesang** [comment]
> Alternatively just a setting to add filter by name.

For the side bar / the "Bazel Build Targets" tree, there is already the setting `bazel.commandLine.queryExpression`.
I guess you could write a Bazel query to filter out `no-ide` tagged targets?

**Comment by theswit** [comment]
Perhaps simpler, can we put the copy/build/run/test option of the `name` target first? I'm hoping if we can't get proper filtering support, maybe just adding a sort change could gain more traction? query expressions may be pretty difficult to get right across a large or complex build codebase. but I suspect most folks care most about the named target itself; I know that's my primary use case.

**Comment by cbandera** [comment]
I think a great solution for this problem was found and implemented with #477 . If you still see the need to filter the results further please comment/reopen the issue.

**Comment by theswit** [comment]
+1, I just noticed this the other day. It was a bit jarring at first, and I wasn't sure if it was related to a new vscode release (1.107) or an extension update. But after the initial 'what the', it's fantastic!
